include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -2982,14 +2982,37 @@ namespace internal
 
 
 
+  /**
+   * Computes the values and derivatives of the 1d polynomials @p poly at the
+   * specified point @p p and stores it in @p shapes.
+   */
+  template <int dim, typename Number>
+  inline void
+  compute_values_of_array(
+    ArrayView<dealii::ndarray<Number, 2, dim>> &        shapes,
+    const std::vector<Polynomials::Polynomial<double>> &poly,
+    const Point<dim, Number> &                          p)
+  {
+    const int n_shapes = poly.size();
+
+    // Evaluate 1d polynomials and their derivatives
+    std::array<Number, dim> point;
+    for (unsigned int d = 0; d < dim; ++d)
```

Comment:
This is a bit ugly, because it simply shuffles around between different formats (`Point<dim, Number>` versus `std::array<Number, dim>`. But I guess the compiler will remove the temporary copy mostly. Actually, it might do even better because it knows that `std::array<> point` cannot alias with the content of `poly[i]`, so it might hoist the loads out of the inner loops for lower degrees.